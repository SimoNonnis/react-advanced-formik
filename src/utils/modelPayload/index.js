import forIn from 'lodash/forIn';
import trim from 'lodash/trim';

function modelPayload(obj) {
  const newObj = {};

  forIn(obj, (value, key) => {
    switch (value) {
      case '':
      case null:
        newObj[key] = null;
        break;
      case true:
        newObj[key] = true;
        break;
      case false:
        newObj[key] = false;
        break;
      case 'true':
        newObj[key] = true;
        break;
      case 'false':
        newObj[key] = false;
        break;
      default:
        newObj[key] = trim(value);
    }
  });

  delete newObj.primary;
  delete newObj.living_at_the_property_you_are_selling_for_at_least_twelve_months;

  return newObj;
}

export default modelPayload;
