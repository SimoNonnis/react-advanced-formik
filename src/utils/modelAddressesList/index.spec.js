/* eslint-env jest */
import modelPayload from './';

const mockPayload = [
  { Id: '52509479.00', StreetAddress: 'P C A Predict', Place: 'Waterside Basin Road Worcester' },
  { Id: '53327590.00', StreetAddress: 'Lock Keepers Cottage Basin Road', Place: 'Worcester' },
  { Id: '26772356.00', StreetAddress: 'Lock View Basin Road', Place: 'Worcester' },
];

describe('Test modelPayload()', () => {
  it('should transform payload array to new modeled array', () => {
    expect(modelPayload(mockPayload)).toMatchSnapshot();
  });
});
