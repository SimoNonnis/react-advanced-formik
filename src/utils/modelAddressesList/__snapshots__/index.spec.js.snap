// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test modelPayload() should transform payload array to new modeled array 1`] = `
Array [
  Object {
    "id": "52509479.00",
    "place": "Waterside Basin Road Worcester",
    "street": "P C A Predict",
  },
  Object {
    "id": "53327590.00",
    "place": "Worcester",
    "street": "Lock Keepers Cottage Basin Road",
  },
  Object {
    "id": "26772356.00",
    "place": "Worcester",
    "street": "Lock View Basin Road",
  },
]
`;
